### PR TITLE
Onboarding storage for autogeneration

### PR DESCRIPTION
**Pull request by leni-msft**
```
> azure-schema-generator@1.0.0 generate-single /home/leni/workspace/azure-resource-manager-schemas/generator
> ts-node cmd/generatesingle "storage/resource-manager"

[/tmp/schm_azspc] executing: git fsck
dangling commit 67fffef061abc631e04d6dc20d6a09362fbd42a3
[/tmp/schm_azspc] executing: git reset -q .
[/tmp/schm_azspc] executing: git checkout -q -- .
[/tmp/schm_azspc] executing: git clean -q -fd
[/tmp/schm_azspc] executing: git gc
[/tmp/schm_azspc] executing: git fetch -q
[/tmp/schm_azspc] executing: git checkout -q origin/master
Using autogenlist config:
{
  "basePath": "storage/resource-manager",
  "namespace": "Microsoft.Storage"
}
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6274 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/l8edzvfubwj --tag=all-api-versions --api-version=2015-05-01-preview --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/storage/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.9.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6274/node_modules/@autorest/core/dist' (3.0.6274)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[38.59 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2015-05-01-preview/Microsoft.Storage.json
Provider Namespace: Microsoft.Storage
API Version: 2015-05-01-preview
Resource Types (Resource Group Scope):
- storageAccounts
================================================================================================================================
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6274 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/oid3hffcf4s --tag=all-api-versions --api-version=2018-03-01-preview --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/storage/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.9.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6274/node_modules/@autorest/core/dist' (3.0.6274)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
WARNING: Found duplicate definition for variable-named type Microsoft.Storage/storageAccounts/managementPolicies. Skipping definition with path '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}/managementPolicies/{managementPolicyName}'.
WARNING: Found duplicate definition for variable-named type Microsoft.Storage/storageAccounts/managementPolicies. Using definition with path '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}/managementPolicies/{managementPolicyName}'.
[85.05 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2018-03-01-preview/Microsoft.Storage.json
Provider Namespace: Microsoft.Storage
API Version: 2018-03-01-preview
Resource Types (Resource Group Scope):
- storageAccounts
- storageAccounts/blobServices/containers
- storageAccounts/blobServices/containers/immutabilityPolicies
- storageAccounts/managementPolicies
================================================================================================================================
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6274 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/gzykrq91ulw --tag=all-api-versions --api-version=2015-06-15 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/storage/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.9.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6274/node_modules/@autorest/core/dist' (3.0.6274)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[36.18 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2015-06-15/Microsoft.Storage.json
Provider Namespace: Microsoft.Storage
API Version: 2015-06-15
Resource Types (Resource Group Scope):
- storageAccounts
================================================================================================================================
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6274 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/wk2avlc9rlk --tag=all-api-versions --api-version=2016-01-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/storage/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.9.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6274/node_modules/@autorest/core/dist' (3.0.6274)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[34.84 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2016-01-01/Microsoft.Storage.json
Provider Namespace: Microsoft.Storage
API Version: 2016-01-01
Resource Types (Resource Group Scope):
- storageAccounts
================================================================================================================================
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6274 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/xry32jxf4dd --tag=all-api-versions --api-version=2016-05-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/storage/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.9.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6274/node_modules/@autorest/core/dist' (3.0.6274)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[35.26 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2016-05-01/Microsoft.Storage.json
Provider Namespace: Microsoft.Storage
API Version: 2016-05-01
Resource Types (Resource Group Scope):
- storageAccounts
================================================================================================================================
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6274 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/m9j7j5q8yl --tag=all-api-versions --api-version=2016-12-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/storage/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.9.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6274/node_modules/@autorest/core/dist' (3.0.6274)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[34.89 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2016-12-01/Microsoft.Storage.json
Provider Namespace: Microsoft.Storage
API Version: 2016-12-01
Resource Types (Resource Group Scope):
- storageAccounts
================================================================================================================================
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6274 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/kqn9e3ei3z --tag=all-api-versions --api-version=2017-06-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/storage/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.9.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6274/node_modules/@autorest/core/dist' (3.0.6274)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[37.72 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2017-06-01/Microsoft.Storage.json
Provider Namespace: Microsoft.Storage
API Version: 2017-06-01
Resource Types (Resource Group Scope):
- storageAccounts
================================================================================================================================
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6274 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/yp5y4xbff --tag=all-api-versions --api-version=2017-10-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/storage/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.9.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6274/node_modules/@autorest/core/dist' (3.0.6274)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[38.83 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2017-10-01/Microsoft.Storage.json
Provider Namespace: Microsoft.Storage
API Version: 2017-10-01
Resource Types (Resource Group Scope):
- storageAccounts
================================================================================================================================
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6274 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/zuqetmzxw7c --tag=all-api-versions --api-version=2018-02-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/storage/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.9.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6274/node_modules/@autorest/core/dist' (3.0.6274)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[93.91 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2018-02-01/Microsoft.Storage.json
Provider Namespace: Microsoft.Storage
API Version: 2018-02-01
Resource Types (Resource Group Scope):
- storageAccounts
- storageAccounts/blobServices/containers
- storageAccounts/blobServices/containers/immutabilityPolicies
================================================================================================================================
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6274 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/6838zjhqdn --tag=all-api-versions --api-version=2018-07-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/storage/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.9.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6274/node_modules/@autorest/core/dist' (3.0.6274)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[89.36 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2018-07-01/Microsoft.Storage.json
Provider Namespace: Microsoft.Storage
API Version: 2018-07-01
Resource Types (Resource Group Scope):
- storageAccounts
- storageAccounts/blobServices
- storageAccounts/blobServices/containers
- storageAccounts/blobServices/containers/immutabilityPolicies
================================================================================================================================
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6274 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/uagrda8ypv --tag=all-api-versions --api-version=2018-11-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/storage/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.9.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6274/node_modules/@autorest/core/dist' (3.0.6274)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[80.56 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2018-11-01/Microsoft.Storage.json
Provider Namespace: Microsoft.Storage
API Version: 2018-11-01
Resource Types (Resource Group Scope):
- storageAccounts
- storageAccounts/blobServices
- storageAccounts/blobServices/containers
- storageAccounts/blobServices/containers/immutabilityPolicies
- storageAccounts/managementPolicies
================================================================================================================================
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6274 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/6aqilgdsy58 --tag=all-api-versions --api-version=2019-04-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/storage/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.9.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6274/node_modules/@autorest/core/dist' (3.0.6274)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[84.62 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2019-04-01/Microsoft.Storage.json
Provider Namespace: Microsoft.Storage
API Version: 2019-04-01
Resource Types (Resource Group Scope):
- storageAccounts
- storageAccounts/blobServices
- storageAccounts/blobServices/containers
- storageAccounts/blobServices/containers/immutabilityPolicies
- storageAccounts/fileServices
- storageAccounts/fileServices/shares
- storageAccounts/managementPolicies
================================================================================================================================
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6274 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/l065lpuwh9s --tag=all-api-versions --api-version=2019-06-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/storage/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.9.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6274/node_modules/@autorest/core/dist' (3.0.6274)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[96.68 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2019-06-01/Microsoft.Storage.json
Provider Namespace: Microsoft.Storage
API Version: 2019-06-01
Resource Types (Resource Group Scope):
- storageAccounts
- storageAccounts/blobServices
- storageAccounts/blobServices/containers
- storageAccounts/blobServices/containers/immutabilityPolicies
- storageAccounts/encryptionScopes
- storageAccounts/fileServices
- storageAccounts/fileServices/shares
- storageAccounts/managementPolicies
- storageAccounts/objectReplicationPolicies
- storageAccounts/privateEndpointConnections
- storageAccounts/queueServices
- storageAccounts/queueServices/queues
- storageAccounts/tableServices
- storageAccounts/tableServices/tables
================================================================================================================================
```
Replaces #973

### CODE REVIEW

**Comment by leni-msft** [comment]
Hi @blueww , according to the mail thread, the PR is regenerated. Please help check.

File: schemas/2019-04-01/deploymentTemplate.json [hunk 1]
```diff
@@ -165,45 +165,6 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2017-03-01/Microsoft.Network.json#/resourceDefinitions/trafficManagerProfiles" },
                   { "$ref": "https://schema.management.azure.com/schemas/2017-05-01/Microsoft.Network.json#/resourceDefinitions/trafficManagerProfiles" },
                   { "$ref": "https://schema.management.azure.com/schemas/2018-04-01/Microsoft.Network.json#/resourceDefinitions/trafficManagerProfiles" },
-                  { "$ref": "https://schema.management.azure.com/schemas/2015-08-01/Microsoft.Storage.json#/resourceDefinitions/storageAccounts" },
```
Comment (blueww) [inline schemas/2019-04-01/deploymentTemplate.json:168]: Why remove storage from this file?

File: schemas/2019-06-01/Microsoft.Storage.json [hunk 2]
```diff
@@ -7,34 +7,22 @@
     "storageAccounts": {
       "type": "object",
       "properties": {
-        "name": {
-          "type": "string",
-          "minLength": 3,
-          "maxLength": 24,
-          "description": "The name of the storage account within the specified resource group. Storage account names must be between 3 and 24 characters in length and use numbers and lower-case letters only."
-        },
-        "type": {
```
Comment (blueww) [inline schemas/2019-06-01/Microsoft.Storage.json:16]: It looks type,tag,sku are missed from storage account? will the template still work after the change?

File: schemas/2019-06-01/Microsoft.Storage.json [hunk 3]
```diff
@@ -99,72 +68,96 @@
...
+          ]
+        },
         "name": {
           "oneOf": [
             {
               "type": "string",
-              "enum": [
-                "default"
-              ]
+              "pattern": "^.*/default$"
```
Comment (blueww) [inline schemas/2019-06-01/Microsoft.Storage.json:153]: Please make sure the name really works.

File: schemas/2019-04-01/deploymentTemplate.json (hunk 1 above)
Comment (leni-msft) [reply schemas/2019-04-01/deploymentTemplate.json:168]: In autogeneration pipeline, the refs are added in autogeneratedResources.json, and then referenced back to this file.

File: schemas/2019-06-01/Microsoft.Storage.json (hunk 3 above)
Comment (leni-msft) [reply schemas/2019-06-01/Microsoft.Storage.json:153]: The test cases have passed. See https://github.com/Azure/azure-resource-manager-schemas/pull/1130/files#diff-e90beca61fcb273a8c3d021d52534605

**Review by blueww — APPROVED** [review]

**Comment by leni-msft** [comment]
@anthony-c-martin @apclouds pls help check & merge

**Comment by leni-msft** [comment]
@anthony-c-martin @jorgecotillo could you help merge this PR?

**Pull request merged**